cheatsheets/gleam-for-typescript-users.md

```diff
@@ -0,0 +1,827 @@
+```
+
+Comments starting with `///` are used to document the following function,
+constant, or type definition. Comments starting with `////` are used to
+document the current module.
+
+```gleam
+//// This module is very important.
+
+/// The answer to life, the universe, and everything.
+const answer: Int = 42
+
+/// A main function
+fn main() {}
+
+/// A Dog type
+type Dog {
+  Dog(name: String, cuteness: Int)
+}
+```
+
+`//` comments are not used while generating documentation files, while
+`////` and `///` will appear in them.
+## Variables
+
+You can rebind variables in both languages.
+
+### TypeScript
+
+```ts
+let a = 50;
+a = a + 100;
+a = 1;
+```
+
+### Gleam
+
+Gleam also has the `let` keyword before its variable names. However, it needs to be used explicitly each time when rebinding/reassinging a new value.
+
+```gleam
+let size = 50
+let size = size + 100
+let size = 1
+```
+## Constants
+
+### TypeScript
+
+In TypeScript, `const` is used for variables that may not be reassigned.
+
+```ts
+const example = 50;
+example = 100; // throws "TypeError: Assignment to constant variable."
+```
+
+### Gleam
+
+In Gleam constants can also be created using the `const` keyword. Constants must be literal values, defined at the top level of a module and functions cannot be used in their definitions.
+
+```gleam
+// the_question.gleam module
+const the_answer = 42
+
+pub fn main() {
+  the_answer
+}
+```
+
+They can also be marked public via the `pub` keyword and will then be
+automatically exported.
+
+```gleam
+pub const the answer = 42
+```
+## Variables type annotations
+
+### TypeScript
+
+TypeScript has optional type annotations (depending on compiler configuration).
+
+```ts
+const example1: number[] = [1, 2, 3];
+const example2: Array<number> = [1, 2, 3]; // equivalent to example 1
+const example3: [number, number, number] = [1, 2, 3];
+const example4 = [1, 2, 3]; // valid, inferred without explicit type annotation
+```
+
+### Gleam
+
+In Gleam type annotations can optionally be given when binding variables.
+
+```gleam
+let some_list: List(Int) = [1, 2, 3]
+let some_string: String = "Foo"
+let another_string = "Bar" // valid, inferred without explicit type annotation
+```
+
+Gleam will check the type annotation to ensure that it matches the type of the
+assigned value. It does not need annotations to type check your code, but you
+may find it useful to annotate variables to hint to the compiler that you want
+a specific type to be inferred.
+## Data types
+
+### Strings
+
+In TypeScript (JavaScript), strings are represented fundamentally as sequences
+of UTF-16 code units. They can be defined with single quotes, double quotes,
+or back-ticks (also known as template strings). Interpolation is supported in
+template literal strings.
+
+In Gleam all strings are UTF-8 encoded binaries. Gleam strings do not allow
+interpolation, yet. Gleam however offers a `string_builder` via its standard
+library for performant string building.
```

Comment:
This representation info is only true on the Erlang platform, and let's remove the word "yet" as it implies it is planned.

Reply:
```suggestion
In Gleam strings are UTF-8 encoded binaries (on the Erlang platform). Gleam strings do not allow
interpolation. Gleam however offers a `string_builder` via its standard
library for performant string building.
```